core/corerepo/stat.go

```diff
@@ -45,11 +46,29 @@ func RepoStat(n *core.IpfsNode, ctx context.Context) (*Stat, error) {
 		return nil, err
 	}
 
+	return &Stat{
+		NumObjects: count,
+		RepoSize:   sizeStat.RepoSize,
+		StorageMax: sizeStat.StorageMax,
+		RepoPath:   path,
+		Version:    fmt.Sprintf("fs-repo@%d", fsrepo.RepoVersion),
+	}, nil
+}
+
+// RepoSize returns a *Stat object with the RepoSize and StorageMax fields set.
+func RepoSize(ctx context.Context, n *core.IpfsNode) (*Stat, error) {
```

Comment:
This feels a bit funky... what about:

```go
type Size struct {
	RepoSize   uint64
	StorageMax uint64
}

type Stat struct {
	StorageSize
	NumObjects  uint64
	RepoPath    string
	Version     string
}
```

And then have `RepoSize` return `Size` (and, really, these functions should return be returning these by value). This would also avoid sending zero values for the missing fields.

Thoughts?

Reply:
yes this is better